src/main/java/com/mitchelltford/discordbot/GatewayDiscordClientConfig.java

```diff
@@ -0,0 +1,18 @@
+package com.mitchelltford.discordbot;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Bean;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.DiscordClientBuilder;
+
+
+@Configuration
+public class GatewayDiscordClientConfig {
+
+  @Bean
+  public GatewayDiscordClient gatewayDiscordConfig(@Value("${bot.token}") String token) {
```

Comment:
My bad, I mistyped this in the issue
```suggestion
  public GatewayDiscordClient gatewayDiscordClientConfig(@Value("${bot.token}") String token) {
```